File: 프로그래머스/LV2/코딩테스트 연습/연습문제/택배상자/택배상자.py
def solution(order):
    stack = []
    order_pos = 0
    for i in range(1, len(order)+1):
        stack.append(i)
        while stack[-1] == order[order_pos]:
            order_pos += 1
            stack.pop()
            if len(stack) == 0:
                break
    return order_pos

File: 프로그래머스/LV2/코딩테스트 연습/연습문제/택배상자/택배상자.cpp
#include <string>
#include <vector>
#include <stack>

using namespace std;

int solution(vector<int> order) 
{
    int answer = 0;
    stack<int> stack;

    for(int i = 1; i <= order.size(); i++)
    {
        stack.push(i);
        while (stack.top() == order[answer])
        {
            answer++;
            stack.pop();
            if(stack.size() == 0)break;
        }
    }
    return answer;
}